CONTRIBUTING.md

```diff
@@ -22,21 +22,86 @@ into three categories:
+
+Once you have those prerequisites installed, simply clone the cuML repo and run
+the `quick_dev.sh` script from the root of the cloned repository:
+```bash
+git clone git@github.com:rapidsai/cuml.git
+cd cuml
+./quick_dev.sh
+```
+The first time you run this, it may take some time to download the development
+Docker image. The script will then
+[mount](https://docs.docker.com/storage/bind-mounts/) the cuML source directory
+into the container and provide you with a prompt from within the container. You
+can edit source files as you would normally would in the `cuml` directory on
+your machine and then build from within the container by running `./build.sh`
+from the container prompt.
+
+The development container can be further configured with the following
+environment variables, if desired:
+
+* Basic Options
+    - `CUML_OS`: The name of one of the supported operating systems for cuML's
+      development containers (Default: `ubuntu18.04`)
+    - `CUML_CUDA`: The desired CUDA version (Default: `11.0`)
+    - `CUML_PY`: The desired Python version (Default: `3.8`)
+    - `CUDA_VISIBLE_DEVICES`: The GPU(s) that will be visible in the dev
+      container (Default: `all`)
+* Advanced Options
+    - `CUML_BUILD_VOLUME`: If set to an absolute path, then this path will be
+      bind-mounted to cuML's build directory, allowing you to persist build
+      artifacts between sessions. If a Docker volume name is used instead, the
+      build directory will be persisted in a volume. This option is
+      particularly useful for those who will be working on more extensive or
+      complex changes.
+    - `CUML_CCACHE_VOLUME`: If set to an absolute path, then this path will be
+      bind-mounted to the `ccache` cache directory within the container,
+      allowing you to persist the cache between sessions. If a Docker volume
+      name is used instead, the cache will be persisted in a volume. This
+      option is useful for those who will be working on multiple branches or
+      building both release and debug versions. You must use the `--ccache`
+      argument to `build.sh` in order to take advantage of this option.
+    - `CUML_VERSION`: The cuML version of the development Docker container.
+      Note that this will NOT affect what version of cuML you currently have
+      checked out in your repo, only the version of the development container.
+      In general, you should not change this option. Defaults to the latest
+      version under development.
+
+With both `CUML_BUILD_VOLUME` and `CUML_CCACHE_VOLUME` set, the development
+container can even be used for long-term development work. E.g.:
+```bash
+CUML_BUILD_VOLUME=cuml-build CUML_CCACHE_VOLUME=cuml-ccache ./quick_dev.sh
+```
+
+#### Conda environment
+If you would rather *not* use the Docker development environment, you can
+follow the instructions in
+[BUILD.md](https://github.com/rapidsai/cuml/blob/main/BUILD.md) to set up a
+conda development environment instead.
```

Comment:
I was thinking it might be worth to add a quick guide here before pointing to the full set, something like:

```
1. Install miniconda if not installed (link to https://docs.conda.io/projects/conda/en/latest/user-guide/install/linux.html)
2. clone cuml 
3. cd cuml
4. conda env create path/to/dev/env
5. conda activate dev_env
6. ./build.sh
```
And then link to the full document for more advanced options, what do you think?